h2o-r/tests/testdir_algos/stackedensemble/runit_stackedensemble_nfolds.R

```diff
@@ -0,0 +1,115 @@
+setwd(normalizePath(dirname(R.utils::commandArgs(asValues=TRUE)$"f")))
+source("../../../scripts/h2o-r-test-setup.R")
+
+stackedensemble.nfolds.test <- function() {
+
+  # This test checks the following (for binomial classification):
+  #
+  # 1) That h2o.stackedEnsemble `metalearner_nfolds` works correctly
+  # 2) That h2o.stackedEnsemble `metalearner_fold_assignment` works correctly
+  # 3) That Stacked Ensemble cross-validation metrics are correctly copied from metalearner
+
+  train <- h2o.uploadFile(locate("smalldata/testng/higgs_train_5k.csv"),
+                          destination_frame = "higgs_train_5k")
+  test <- h2o.uploadFile(locate("smalldata/testng/higgs_test_5k.csv"),
+                         destination_frame = "higgs_test_5k")
+  y <- "response"
+  x <- setdiff(names(train), y)
+  train[,y] <- as.factor(train[,y])
+  test[,y] <- as.factor(test[,y])
+  nfolds <- 5
+
+  # Train & Cross-validate a GBM
+  my_gbm <- h2o.gbm(x = x,
+                    y = y,
+                    training_frame = train,
+                    distribution = "bernoulli",
+                    ntrees = 10,
+                    nfolds = nfolds,
+                    keep_cross_validation_predictions = TRUE,
+                    seed = 1)
+
+  # Train & Cross-validate a RF
+  my_rf <- h2o.randomForest(x = x,
+                            y = y,
+                            training_frame = train,
+                            ntrees = 10,
+                            nfolds = nfolds,
+                            keep_cross_validation_predictions = TRUE,
+                            seed = 1)
+
+
+  # Check that not setting nfolds still produces correct results
+  stack0 <- h2o.stackedEnsemble(x = x,
+                                y = y,
+                                training_frame = train,
+                                base_models = list(my_gbm, my_rf))
+  # Check that metalearner_nfolds is correctly stored in model output
+  expect_equal(stack0@parameters$metalearner_nfolds, NULL)
+  expect_equal(stack0@allparameters$metalearner_nfolds, 0)
+  # Check that the metalearner was cross-validated with the correct number of folds
+  meta0 <- h2o.getModel(stack0@model$metalearner$name)
+  expect_equal(meta0@parameters$nfolds, NULL)
+  expect_equal(meta0@allparameters$nfolds, 0)
```

Comment:
Why the diff between `parameters` and `allparameters` again?

Reply:
`parameters` stores only the named parameters set explicitly by the user, where as `allparameters` stores all the parameters and their values (default values with the exception of those you specified explicitly).

Reply:
In this particular instance, it's checking that the metalearner's `nfolds` was not set (NULL) and that the value of the parameter internally is still 0.

Reply:
Ah, okay. Makes sense.